### PR TITLE
Support changing service_name in datadog integration

### PR DESCRIPTION
**Pull request by paneq**
By default datadog will use the service name such as `myservice` which will be the same for web and karafka processes. 

However, it is beneficial to separate those and have  `myservice-karafka` as a separate service name. This allows setting different default operation in Datadog UI and makes analyzing traces much easier.

In https://docs.datadoghq.com/tracing/trace_collection/dd_libraries/ruby/ you can find examples how every integration supports setting service name.

Tested manually in Datadog as well.
image

### CODE REVIEW

**Comment by mensfeld** [comment]
Not a heavy DD user anymore: would it be beneficial to do the same to the second DD listener?

File: lib/karafka/instrumentation/vendors/datadog/logger_listener.rb [hunk 1]
```diff
@@ -12,11 +12,13 @@ class LoggerListener
           include ::Karafka::Core::Configurable
           extend Forwardable
 
-          def_delegators :config, :client
+          def_delegators :config, :client, :service_name
 
           # `Datadog::Tracing` client that we should use to trace stuff
           setting :client
 
+          setting :service_name
```
Comment (mensfeld) [inline lib/karafka/instrumentation/vendors/datadog/logger_listener.rb:20]: is there a default for this value or will the default be used when nil?

**Review by mensfeld — COMMENTED** [review]
One question: is this change backwards compatible? could there be an explicit default that would match the current behaviour?

File: lib/karafka/instrumentation/vendors/datadog/logger_listener.rb (hunk 1 above)
Comment (paneq) [reply lib/karafka/instrumentation/vendors/datadog/logger_listener.rb:20]: default when `nil`:  https://github.com/DataDog/dd-trace-rb/blob/afc4a025c7c8717050901f1419b7b21856aa9a2a/lib/datadog/tracing/tracer.rb#L128

>  is this change backwards compatible? could there be an explicit default that would match the current behaviour?

so it's backwards compatible. The only difference is that now `nil` will be passed explicitly when not set.

**Comment by paneq** [comment]
> would it be beneficial to do the same to the second DD listener?

No idea. Tried to look into ddtrace gem codebase to find out, but I couldn't quickly figure it out. My gut feeling is that yes, it should be possible an beneficial.

**Review by mensfeld — APPROVED** [review]

**Review by mensfeld — APPROVED** [review]

**Comment by mensfeld** [comment]
Retargeting so I can polish it. Once merged to master will be part of the upcoming release since it does not have breaking changes :bowing_man:

**Pull request merged**